great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -295,6 +298,72 @@ def get_batch_list_from_batch_request(
         self.sort_batches(batch_list)
         return batch_list
 
+    def _create_batch_spec_kwargs(self) -> dict[str, Any]:
+        """Creates batch_spec_kwargs used to instantiate a SqlAlchemyDatasourceBatchSpec
+
+        This is called by get_batch_list_from_batch_request to generate the batches.
+
+        Returns:
+            A dictionary that will be passed to SqlAlchemyDatasourceBatchSpec(**returned_dict)
+        """
+        raise NotImplementedError
+
+    def data_from(self) -> str:
+        """Returns the from clause that would be used to query this data
+
+        Returns:
+            The from clause which might be a table name or a query.
+        """
+        raise NotImplementedError
+
+
+class QueryAsset(SQLAsset):
+    # Instance fields
+    type: Literal["query"] = "query"  # type: ignore[assignment]
+    query: str
+
+    @pydantic.validator("query")
+    def query_must_start_with_select(cls, v: str):
+        if not (v.upper().startswith("SELECT") and v[6].isspace()):
```

Comment:
@billdirks I think that this should be flexible in terms of blank spaces, tabs, etc. (please see an earlier comment).  Thanks.

Reply:
Done.

Reply:
:bow